pkg/util/urlpattern/urlpattern.go

```diff
@@ -12,7 +12,7 @@ var InvalidPatternError = errors.New("invalid pattern")
 
 var urlPatternRegex = regexp.MustCompile(`^` +
 	`(?:(\*|git|http|https|ssh)://)` +
-	`(\*|(?:\*\.)?[^/*]+)` +
+	`(\*|(?:\*\.)?[^@/*]+)` +
 	`(/.*)` +
 	`$`)
```

Comment:
We're disallowing @ in the host part of the /pattern/, not in the URL matched.  A pattern of http://example.com/* would still match the URL http://user:password@example.com/ (although I'll add a test case for this).  I want to disable @ in the pattern because I don't think it makes any sense for us to match user:password@ in the URL.